gensim/models/keyedvectors.py

```diff
@@ -1384,12 +1387,40 @@ def init_sims(self, replace=False):
             else:
                 self.vectors_norm = (self.vectors / sqrt((self.vectors ** 2).sum(-1))[..., newaxis]).astype(REAL)
 
+    def relative_cosine_similarity(self, wa, wb, topn=10):
+        """Compute the relative cosine similarity between two words given top-n similar words,
+        proposed by Artuur Leeuwenberg, Mihaela Vela, Jon Dehdari, Josef van Genabith
+        "A Minimally Supervised Approach for Synonym Extraction with Word Embeddings"
+        <https://ufal.mff.cuni.cz/pbml/105/art-leeuwenberg-et-al.pdf>.
+        To calculate relative cosine similarity between two words, equation (1) of the paper is used.
+        For WordNet synonyms, if rcs(topn=10) is greater than 0.10 then wa and wb are more similar than
+        any arbitrary word pairs.
+        Parameters
+        ----------
+        wa: str
+         word for which we have to look top-n similar word.
+        wb: str
+         word for which we evaluating relative cosine similarity with wa.
+        topn: int, optional
+         Number of top-n similar words to look with respect to wa.
+        Returns
+        -------
+        numpy.float64
+            relative cosine similarity between wa and wb.
+        """
+
+        sims = self.similar_by_word(wa, topn)
+        rcs = (self.similarity(wa, wb)) / (sum(result[1] for result in sims))
```

Comment:
Is it guaranteed that `sims` is not empty? Otherwise this line will throw an exception.

Reply:
`sims` is not empty, since `relative_cosine_similarity` is called with arguments `wa,wb` and `topn` and 'wa' is used to compute `sims`.

Reply:
I'm not sure what that means or how that follows, but if a non-obvious condition is known not to happen, it's a good practice to make it explicit: `assert sims`.

This makes the assumptions clearer to humans reading the code, as well as to stop errors silently propagation if the assumption is broken (perhaps due to some future refactoring).

Reply:
@rsdel I suspect there's a corner case – such as when the full vector set has only one word – where `sims` could be empty... but then the failure of `similarity(wa, wb)` should give a meaningful error.

Reply:
Okay, I will do that.

Reply:
I have made the changes.

Reply:
The change seems unrelated to our discussion here. How are `len(self.vocab)` and `topn` connected to whether `sims` is empty or not?

Reply:
If you want to test for empty `sims`, why test for some proxy instead of for empty `sims`?

This check may be useful for other purposes (I don't know), but it is too round-about as an invariant check for empty results / pending division by zero.